e2e/security/security.go

```diff
@@ -0,0 +1,292 @@
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath     string `split_words:"true"`
+	TestDir     string `split_words:"true"`
+	RunDisabled bool   `default:"false"`
+}
+
+var testenv testingEnv
+var imagePath string
+
+// testSecurityUnpriv tests security flag fuctionality for singularity exec without elevated privileges
+func testSecurityUnpriv(t *testing.T) {
+	tests := []struct {
+		name          string
+		image         string
+		action        string
+		expectID      string
+		argv          []string
+		opts          e2e.ExecOpts
+		exit          int
+		expectSuccess bool
+	}{
+		// taget UID/GID
+		{
+			name:          "Set_uid",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"id", "-u"},
+			opts:          e2e.ExecOpts{Security: []string{"uid:99"}},
+			expectID:      "99",
+			exit:          1,
+			expectSuccess: false,
+		},
+		{
+			name:          "Set_gid",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"id", "-g"},
+			opts:          e2e.ExecOpts{Security: []string{"gid:99"}},
+			expectID:      "99",
+			exit:          1,
+			expectSuccess: false,
+		},
+		// seccomp from json file
+		{
+			name:          "SecComp_BlackList",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"mkdir", "/tmp/foo"},
+			opts:          e2e.ExecOpts{Security: []string{"seccomp:./testdata/seccomp-profile.json"}},
+			exit:          1,
+			expectSuccess: false,
+		},
+		{
+			name:          "SecComp_true",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"true"},
+			opts:          e2e.ExecOpts{Security: []string{"seccomp:./testdata/seccomp-profile.json"}},
+			exit:          0,
+			expectSuccess: true,
+		},
+		// capabilities
+		{
+			name:          "capabilities_keep_true",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:          e2e.ExecOpts{KeepPrivs: true},
+			exit:          1,
+			expectSuccess: false,
+		},
+		{
+			name:          "capabilities_keep-false",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:          e2e.ExecOpts{KeepPrivs: false},
+			exit:          1,
+			expectSuccess: false,
+		},
+		{
+			name:          "capabilities_drop",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:          e2e.ExecOpts{DropCaps: "CAP_NET_RAW"},
+			exit:          1,
+			expectSuccess: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("unpriv "+tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			stdout, stderr, exitCode, err := e2e.ImageExec(t, testenv.CmdPath, tt.action, tt.opts, tt.image, tt.argv)
+
+			lines := bytes.Fields([]byte(stdout))
+
+			if tt.expectSuccess && (exitCode != 0) {
```

Comment:
Please consider [testing all cases](https://github.com/sylabs/singularity/wiki/End-to-end-testing#check-all-cases)